File: CartMinder/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/cartminder/activity/MainActivity.java
package id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.R;
import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.fragment.CartsFragment;

public class MainActivity extends AppCompatActivity {

    public MainActivity() {
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onStart() {
        super.onStart();
        CoordinatorLayout layout = (CoordinatorLayout) findViewById(R.id.layout_fragment_container);
        layout.removeAllViewsInLayout();
        Fragment mainFragment = new CartsFragment();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.layout_fragment_container, mainFragment);
        transaction.commit();
    }
}

File: app4_fragment/app/src/main/java/com/example/app4_fragment/MainActivity.kt
package com.example.app4_fragment

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.app4_fragment.fragments.ContentFragment
import com.example.app4_fragment.fragments.MainFragment


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val mainFragment = MainFragment()

        supportFragmentManager.beginTransaction().apply {
            replace(R.id.mainlayout, mainFragment)
            commit()
        }
    }
}